tenable/io/v3/vm/agent_groups.py

```diff
@@ -0,0 +1,178 @@
+
+    def create(self, name: str) -> Dict:
+        '''
+        Creates a new agent group.
+
+        :devportal:`agent-groups: create <agent-groups-create>`
+
+        Args:
+            name (str): The name of the agent group
+
+        Returns:
+            :obj:`dict`:
+                The dictionary object representing the newly minted agent group
+
+        Examples:
+            >>> group = tio.v3.vm.agent_groups.create('New Agent Group')
+        '''
+        return self._post(f'', json={'name': name})
+
+    def delete(self, group_id: int) -> None:
+        '''
+        Delete an agent group.
+
+        :devportal:`agent-groups: delete <agent-groups-delete>`
+
+        Args:
+            group_id (int): The id of the agent group to delete
+
+        Returns:
+            :obj:`None`
+
+        Examples:
+            >>> tio.v3.vm.agent_groups.delete(1)
+        '''
+        self._delete(f'{group_id}')
+
+    def delete_agent(self, group_id: int, *agent_ids: int)\
+            -> Union[None, Dict]:
+        '''
+        Delete one or many agents from an agent group.
+
+        :devportal:`agent-groups: delete-agent <agent-groups-delete-agent>`
+
+        Args:
+            group_id (int): The id of the agent group to remove the agent from
+            *agent_ids (int): The id of the agent to be removed
+
+        Returns:
+            :obj:`dict` or :obj:`None`:
+                If deleting a singular agent, a :obj:`None` response will be
+                returned.  If deleting multiple agents, a :obj:`dict` response
+                will be returned with a Job resource record.
+
+        Examples:
+            Delete a singular agent from an agent group:
+
+            >>> tio.v3.vm.agent_groups.delete_agent(1, 1)
+
+            Delete multiple agents from an agent group:
+
+            >>> tio.v3.vm.agent_groups.delete_agent(1, 1, 2, 3)
+        '''
+        if len(agent_ids) <= 1:
+            # if only a singular agent_id was passed, then we will want to
+            self._delete(f'{group_id}/agents/{agent_ids[0]}')
+        else:
+            # if multiple agent ids were requested to be deleted, then we will
+            # call the bulk deletion API.
+            return self._post(
+                f'{group_id}/agents/_bulk/remove',
+                json={'items': [i for i in agent_ids]}
+            )
+
+    def details(self):
+        NotImplemented('This method will be updated later'
+                       'once the filter API will develop')
+
+    def search(self):
+        NotImplemented('Search and Filter functionality '
+                       'will be updated later.')
+
+    def task_status(self, group_id: int, task_uuid: str) -> Dict:
+        '''
+        Retrieves the current status of a bulk task.
+
+        :devportal:`bulk-operations: bulk-agent-group-status
+        <bulk-task-agent-group-status>`
+
+        Args:
+            group_id (int): The id of the group
```

Comment:
group_id will now be of type UUID (str) as per latest v3 spec.

Reply:
done,